app/src/main/java/com/bnyro/translate/ui/views/EnginePref.kt

```diff
@@ -89,11 +90,17 @@ fun EnginePref() {
         }
 
         if (engine.apiKeyState != ApiKeyState.DISABLED) {
+            val keyLabelResId = if (engine is KagiEngine) {
```

Comment:
I'd prefer to not have engine specific stuff here, rather add an enum `apiKeyType` to the base `TranslationEngine` class, with the values `API_KEY` and `SESSION_TOKEN`